blocks/algorithms/__init__.py

```diff
@@ -164,17 +167,20 @@ class GradientDescent(DifferentiableCostMinimizer):
     """
     def __init__(self, step_rule=None, gradients=None, **kwargs):
         super(GradientDescent, self).__init__(**kwargs)
-        self.gradients = (
-            gradients if gradients
-            else dict(
-                zip(self.params, tensor.grad(self.cost, self.params))))
+        self.gradients = gradients
+        if not self.gradients:
+            logger.info("Taking the cost gradient")
```

Comment:
Actually taking the gradient can take more then half-a-minute when you are working with scan. So you want this logging to be always on, and you do no want to see those really detailed messages you get at the debug level.